flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestServerEndpoint.java

```diff
@@ -181,14 +184,44 @@ protected void initChannel(SocketChannel ch) {
 				.channel(NioServerSocketChannel.class)
 				.childHandler(initializer);
 
-			log.debug("Binding rest endpoint to {}:{}.", restBindAddress, restBindPort);
-			final ChannelFuture channel;
-			if (restBindAddress == null) {
-				channel = bootstrap.bind(restBindPort);
-			} else {
-				channel = bootstrap.bind(restBindAddress, restBindPort);
+			ChannelFuture channel;
+
+			// parse port range definition and create port iterator
+			Iterator<Integer> portsIterator;
+			try {
+				portsIterator = NetUtils.getPortRangeFromString(restBindPort);
+			} catch (Exception e) {
```

Comment:
`getPortRangeFromString ` also throws `IllegalConfigurationException` which is a unchecked exception defined in `flink-core`, so here we catch a general `Exception` and in the catch block then rethrow a unified `IllegalArgumentException `. This code style used in `BootstrapTools#startActorSystem`

Reply:
Here, we catch `Exception` so that regardless of `NumberFormatException` or `IllegalConfigurationException` will be caught, isn't it? Or did I misunderstand what you mean? Do you want me to catch a more specific exception?

Reply:
What I meant is "should we catch `RuntimeExceptions`". function signature of `startActorSystem` explicitly throws a generic `Exception` so there's no way to distinguish in `BootstrapTool` use case. But here since we are directly using `NetUtil.getPor...`, we might be able to do a better job. 

Regarding whether we should catch it. I am fine with either decision as long as they are properly documented.

Reply:
OK, thank you for your clarification. It seems that I misunderstood what you mean. I thought you meant to capture Exception and could not catch IllegalConfigurationException. Now it seems that what you mean is that we should break down the capture of a particular exception. Considering that `NetUtils.getPortRangeFromString` will throw at least two possible exceptions, one checked and one unchecked. I think we capture Exception directly.

Reply:
Why catch anything in the first place? The wrapped exception is not adding any value.

Reply:
hmm, yes, the generic Exception cannot provide more detailed information.